src/Tty-macos.zig

```diff
@@ -0,0 +1,313 @@
+                        vx.postEvent(.focus_out);
+                    }
+                },
+                .paste_start => {
+                    if (@hasField(Event, "paste_start")) {
+                        vx.postEvent(.paste_start);
+                    }
+                },
+                .paste_end => {
+                    if (@hasField(Event, "paste_end")) {
+                        vx.postEvent(.paste_end);
+                    }
+                },
+                .cap_kitty_keyboard => {
+                    log.info("kitty keyboard capability detected", .{});
+                    vx.caps.kitty_keyboard = true;
+                },
+                .cap_kitty_graphics => {
+                    if (!vx.caps.kitty_graphics) {
+                        log.info("kitty graphics capability detected", .{});
+                        vx.caps.kitty_graphics = true;
+                    }
+                },
+                .cap_rgb => {
+                    log.info("rgb capability detected", .{});
+                    vx.caps.rgb = true;
+                },
+                .cap_unicode => {
+                    log.info("unicode capability detected", .{});
+                    vx.caps.unicode = true;
+                    vx.screen.unicode = true;
+                },
+                .cap_da1 => {
+                    std.Thread.Futex.wake(&vx.query_futex, 10);
+                },
+            }
+        }
+    }
+}
+
+/// write to the tty. These writes are buffered and require calling flush to
+/// flush writes to the tty
+pub fn write(self: *Tty, bytes: []const u8) !usize {
+    return self.buffered_writer.write(bytes);
+}
+
+/// flushes the write buffer to the tty
+pub fn flush(self: *Tty) !void {
+    try self.buffered_writer.flush();
+}
+
+/// makeRaw enters the raw state for the terminal.
+pub fn makeRaw(fd: os.fd_t) !os.termios {
+    const state = try os.tcgetattr(fd);
+    var raw = state;
+    // see termios(3)
+    raw.iflag.IGNBRK = false;
+    raw.iflag.BRKINT = false;
+    raw.iflag.PARMRK = false;
+    raw.iflag.ISTRIP = false;
+    raw.iflag.INLCR = false;
+    raw.iflag.IGNCR = false;
+    raw.iflag.ICRNL = false;
+    raw.iflag.IXON = false;
+    raw.iflag.IUTF8 = true;
+
+    raw.oflag.OPOST = false;
+
+    raw.lflag.ECHO = false;
+    raw.lflag.ECHONL = false;
+    raw.lflag.ICANON = false;
+    raw.lflag.ISIG = false;
+    raw.lflag.IEXTEN = false;
+
+    raw.cflag.CSIZE = .CS8;
+    raw.cflag.PARENB = false;
+
+    raw.cc[@intFromEnum(std.posix.V.MIN)] = 1;
+    raw.cc[@intFromEnum(std.posix.V.TIME)] = 0;
+    try os.tcsetattr(fd, .FLUSH, raw);
+
+    return state;
+}
+
+/// The size of the terminal screen
+pub const Winsize = @import("Tty.zig").Winsize;
+
+fn getWinsize(fd: os.fd_t) !Winsize {
+    var winsize = os.winsize{
+        .ws_row = 0,
+        .ws_col = 0,
+        .ws_xpixel = 0,
+        .ws_ypixel = 0,
+    };
+
+    const TIOCGWINSZ = 1074295912;
```

Comment:
This constant should be available through `std.os.T`, was it not working for darwin?

Reply:
ahhh, i was throwing many things at the wall and saw that ghosttyy just writes this. it works just fine, so i'll delete